util/auto_skip_compressor.cc

```diff
@@ -0,0 +1,111 @@
+//  Copyright (c) Meta Platforms, Inc. and affiliates.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+//
+
+#include "util/auto_skip_compressor.h"
+
+#include "options/options_helper.h"
+#include "rocksdb/advanced_compression.h"
+#include "util/random.h"
+namespace ROCKSDB_NAMESPACE {
+
+int CompressionRejectionProbabilityPredictor::Predict() const {
+  return pred_rejection_prob_percentage_;
+}
+
+size_t CompressionRejectionProbabilityPredictor::attempted_compression_count()
+    const {
+  return rejected_count_ + compressed_count_;
+}
+
+bool CompressionRejectionProbabilityPredictor::Record(
+    Slice uncompressed_block_data, std::string* compressed_output,
+    const CompressionOptions& opts) {
+  if (compressed_output->size() >
+      (static_cast<uint64_t>(opts.max_compressed_bytes_per_kb) *
+       uncompressed_block_data.size()) >>
+      10) {
+    rejected_count_++;
+  } else {
+    compressed_count_++;
+  }
+  if (attempted_compression_count() >= window_size_) {
+    pred_rejection_prob_percentage_ = static_cast<int>(
+        rejected_count_ * 100 / (compressed_count_ + rejected_count_));
+    compressed_count_ = 0;
+    rejected_count_ = 0;
+    assert(attempted_compression_count() == 0);
+  }
+  return true;
+}
+AutoSkipCompressorWrapper::AutoSkipCompressorWrapper(
+    std::unique_ptr<Compressor> compressor, const CompressionOptions& opts,
+    const CompressionType type)
+    : CompressorWrapper::CompressorWrapper(std::move(compressor)),
+      opts_(opts),
+      type_(type),
+      predictor_(
+          std::make_shared<CompressionRejectionProbabilityPredictor>(10)) {
+  (void)type_;
+  (void)opts_;
```

Comment:
why do you need to (void)opts_? Doesn't CompressBlockAndRecord() needs it?

Reply:
ping ping @shubhajeet - it seems that the merged in PR has not addressed some of the comments.